cgi-bin/DW/Media.pm

```diff
@@ -110,9 +110,15 @@ sub upload_media {
 
     # set the security
     my $sec = $opts{security} || 'public';
+    $sec = 'usemask' if $sec =~ /^(?:friends|access)$/;
```

Comment:
(If I really wanted to go on a rampage I'd figure out a way to abstract this out so we don't keep writing the same code wrongly everywhere, but I think there are other things that would be a better use of my time.)